src/stores/CatalogOnlineQuery/CatalogOnlineQueryConfigStore.ts

```diff
@@ -310,13 +311,25 @@ export class CatalogOnlineQueryConfigStore {
         return tables;
     }
 
-    convertToDeg(pixelCoords: Point2D, system?: SystemType): {x: string | undefined; y: string | undefined} {
+    private isValidPrecisionString(precision: string): boolean {
+        return precision === "*" || /^[0-9]+$/.test(precision);
+    }
+
+    private getEffectivePrecision(precision?: string): string {
+        if (precision && !this.isValidPrecisionString(precision)) {
+            console.warn(`Invalid precision string: ${precision}. Expected '*' or a non-negative integer. Using default precision *.`);
+        }
+        const overlay = AppStore.Instance.overlaySettings;
+        const effectivePrecision = precision ?? (overlay.numbers.customPrecision ? overlay.numbers.precision.toString() : "*");
+        return this.isValidPrecisionString(effectivePrecision) ? effectivePrecision : "*";
```

Comment:
I prefer call isValidPrecisionString only once.
```suggestion
if (precision && this.isValidPrecisionString(precision)) {
    return precision;
} else {
    if (precision) {
        console.warn(`Invalid precision string: ${precision}. Using default precision *.`);
    }
    const overlay = AppStore.Instance.overlaySettings;
    return overlay.numbers.customPrecision ? overlay.numbers.precision.toString() : "*";
}
```

Reply:
Done. Thanks!